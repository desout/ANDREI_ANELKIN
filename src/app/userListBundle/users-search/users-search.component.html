<div class="searchComponent">
  <div class="input-group">
  <input type="text" placeholder="Search" class="form-control" (keyup)="search(searchInput.value)" #searchInput>
    <button  type="button"  (click)="search(searchInput.value)" class="btn btn-success material-icons">search</button>

  </div>
  <div *ngIf="searching === true" class="loader"></div>
</div>
