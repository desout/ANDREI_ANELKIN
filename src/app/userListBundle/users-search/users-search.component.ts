import {Component, Inject, OnInit} from '@angular/core';
import {debounceTime, switchMap, tap} from 'rxjs/operators';
import {UserListViewComponent} from '../user-list-view/user-list-view.component';
import { Subject} from 'rxjs';
import {UserDropDownService} from '../../user-dropdown.service';

@Component({
  selector: 'app-users-search',
  templateUrl: './users-search.component.html',
  styleUrls: ['./users-search.component.scss']
})

export class UsersSearchComponent implements OnInit {
  searching = false;
  public searchUsersTerm = new Subject<string>();
  constructor(@Inject(UserListViewComponent) private parent: UserListViewComponent,
              private userDropDownService: UserDropDownService) {
      this.searchUsersTerm
      .pipe(
        debounceTime(500),
        tap(() => this.searching = true),
        switchMap((term: string) => {
          this.userDropDownService.term = term;
          return this.userDropDownService.getFilteredUsers();
        }),
        tap(() => this.searching = false))
        .subscribe((users) => this.userDropDownService.users = users);
  }
  ngOnInit() {
    this.search('');
  }
  search(term: string) {
    this.searchUsersTerm.next(term);
  }
}

