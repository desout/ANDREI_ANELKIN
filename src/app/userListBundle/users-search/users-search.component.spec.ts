import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { UsersSearchComponent } from './users-search.component';
import {NO_ERRORS_SCHEMA} from '@angular/core';
import {TestModule} from '../../Test.module';

describe('UsersSearchComponent', () => {
  let component: UsersSearchComponent;
  let fixture: ComponentFixture<UsersSearchComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [],
      schemas: [NO_ERRORS_SCHEMA],
      imports: [TestModule],
      providers: []
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(UsersSearchComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
