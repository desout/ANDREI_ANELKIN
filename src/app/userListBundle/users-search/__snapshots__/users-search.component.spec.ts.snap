// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersSearchComponent should create 1`] = `
<div
  id="root0"
  ng-version="7.0.3"
>
  <div
    class="searchComponent"
  >
    <div
      class="input-group"
    >
      <input
        class="form-control"
        placeholder="Search"
        type="text"
      />
      <button
        class="btn btn-success material-icons"
        type="button"
      >
        search
      </button>
    </div>
    
  </div>
</div>
`;
